readthedocs/projects/tasks/search.py

```diff
@@ -120,7 +121,36 @@ def collect(self, sync_id: int):
         self.version.imported_files.exclude(build=sync_id).delete()
 
 
-def _get_indexers(*, version, search_ranking, search_ignore, search_index_name=None):
+class FileManifestIndexer(Indexer):
+    def __init__(self, version: Version, build: Build):
+        self.version = version
+        self.build = build
+        self._hashes = {}
+
+    def process(self, html_file: HTMLFile, sync_id: int):
+        self._hashes[html_file.path] = html_file.processed_json["main_content_hash"]
+
+    def collect(self, sync_id: int):
+        manifest = {
+            "build": {
+                "id": self.build.id,
+            },
+            "files": {
+                path: {
+                    "hash": hash,
```

Comment:
I would probably use `main_content_hash` here as well since it's more explicit. This will avoid confusing it with the whole file hash, and leave us the door open to the future in case we need to the whole file hash as well.

```suggestion
                    "main_content_hash": hash,
```